bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+import logging
+
+from todoist_api_python.api import TodoistAPI
+import typing_extensions
+
+from datetime import datetime, time
+
+from bugwarrior import config
+from bugwarrior.services import Service, Issue, Client
+
+log = logging.getLogger(__name__)
+
+
+class TodoistConfig(config.ServiceConfig):
+    service: typing_extensions.Literal["todoist"]
+    token: str
+    filter: str = None
+    char_open_bracket: str = "〈"
+    char_close_bracket: str = "〉"
+    inline_links: bool = True
+
+
+class TodoistClient(Client):
+    def __init__(self, token, filter):
+        self._api = TodoistAPI(token)
+        self.filter = filter
+
+    def get_projects(self):
+        all_projects = []
+        projects_iter = self._api.get_projects()
+        for projects in projects_iter:
+            for project in projects:
+                all_projects.append(project)
+        return all_projects
+
+    def get_sections(self):
+        all_sections = []
+        sections_iter = self._api.get_sections()
+        for sections in sections_iter:
+            for section in sections:
+                all_sections.append(section)
+        return all_sections
+
+    def get_users(self, project_id):
+        all_users = []
+        users_iter = self._api.get_collaborators(project_id)
+        for users in users_iter:
+            for user in users:
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        return tasks_iter
+
+
+class TodoistIssue(Issue):
+    ASSIGNEE = "todoistassignee"
+    ASSIGNER = "todoistassigner"
+    CONTENT = "todoistcontent"
+    DESCRIPTION = "todoistdescription"
+    DURATION = "todoistduration"
+    ID = "todoistid"
+    SECTION = "todoistsection"
+    URL = "todoisturl"
+
+    PRIORITY_MAP = {
+        4: "H",
+        3: "M",
+        2: "L",
+        1: None,
+    }
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Todoist ID",
+        },
+        CONTENT: {
+            "type": "string",
+            "label": "Todoist Content",
+        },
+        DESCRIPTION: {
+            "type": "string",
+            "label": "Todoist Description",
+        },
+        DURATION: {
+            "type": "string",
+            "label": "Todoist Duration",
+        },
+        SECTION: {
+            "type": "string",
+            "label": "Todoist Section",
+        },
+        ASSIGNEE: {
+            "type": "string",
+            "label": "Todoist Assignee",
+        },
+        ASSIGNER: {
+            "type": "string",
+            "label": "Todoist Assigner",
+        },
+        URL: {
+            "type": "string",
+            "label": "Todoist URL",
+        },
+    }
+
+    UNIQUE_KEY = (ID, ID)
+
+    # replace characters that cause escaping issues in teh description like [] and "
```

Comment:
This eliminates a typo but it also eliminates the implication that this only impacts the description when it is also applied to the content.

```suggestion
    # replace characters that cause escaping issues like [] and "
```